infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/LogReplicationSourceManager.java

```diff
@@ -61,119 +70,89 @@
      */
     private final LogReplicationConfig config;
 
+    /*
+     * Log Replication MetadataManager.
+     */
+    private final LogReplicationMetadataManager metadataManager;
+
+    /*
+     * Last ack'd timestamp from Receiver
+     */
+    private AtomicLong lastAckedTimestamp = new AtomicLong(Address.NON_ADDRESS);
+
+    /*
+     * Periodic Thread which reads the last Acked Timestamp and writes it to the metadata table
+     */
+    ScheduledExecutorService lastAckedTsPoller;
+
+    /*
+     * Interval at which the thread reads the last Acked Timestamp
+     */
+    private static int ACKED_TS_READ_INTERVAL_SECONDS = 15;
```

Comment:
no need to do this for this review, but we might add an issue to track the fact that we should move this configuration out to the config file.